### PR TITLE
General clean-up and stabilization of the solution

### PR DESCRIPTION
**Pull request by carstendietrich**
- Update Readme's to reflect current configs / fix typos
- Update CueConfig to match used configurations
- Remove unused / commented out code
- Add module_tests
- Fix non-working SearchBy implementation
- Fix non-working Bleve pagination
- Add config for additional Facets/Sorting
- Allow different CSV delimiter via config
- Add support for stock level / saleable / retailer name attribute
- Add active Facets to search response
- Avoid indexing foreign language attributes
- Remove default title sort option

Todo:
- [x] Fix tests
- [x] Update Changelog

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,45 +1,44 @@
 [![Build Status](https://travis-ci.org/i-love-flamingo/flamingo-commerce-adapter-standalone.svg?branch=master)](https://travis-ci.org/i-love-flamingo/flamingo-commerce-adapter-standalone?branch=master)
 
+# Flamingo Commerce Adapters Standalone
 
-# Flamingo Commerce Adapters Standalone 
+This repository contains modules that allow to run Flamingo Commerce in a standalone mode.
+(Not connected to any thirds party headless ecommerce API).
```
Comment (el0c1n) [inline Readme.md]:
```suggestion
(Not connected to any third party headless ecommerce API).
```

File: Readme.md [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
-* csvindexing
-    * usful addon module for the *commercesearch* module
-    * Provide a "IndexUpdater" that loads products and categories from a CSV file
-    * See [Module csvcommerce Readme](csvindexing/Readme.md)
+* **commercesearch**
+  * Provide Product; ProductSearch; Category and Search. That means you can have working product views, category views and search features.
+  * The module therefore internally uses Repository where products are stored and received from. You can choose between
+    * A simple "InMemory" version
+    * An implementation based on bleve - a go based indexed search implementation ( https://github.com/blevesearch/bleve)
+  * The module requires someone who takes care if indexing new products and categories. Therefore, it expects an implementation of an "IndexUpdater"
```
Comment (el0c1n) [inline Readme.md]:
```suggestion
  * The module requires someone who takes care of indexing new products and categories. Therefore, it expects an implementation of an "IndexUpdater"
```

File: Readme.md [hunk 3]
```diff
@@ -1,45 +1,44 @@
...
+  * See [Module commercesearch Readme](commercesearch/Readme.md)
     
-* emailplaceorder
-    * Like the name says a module that just sends mails (to customer and storeowner) after placing an order
+* **csvindexing**
+  * usful addon module for the *commercesearch* module
```
Comment (el0c1n) [inline Readme.md]:
```suggestion
  * useful addon module for the *commercesearch* module
```

File: commercesearch/Readme.md [hunk 4]
```diff
@@ -3,59 +3,69 @@
...
 
 The provided adapters are:
- * productService (to retrieve single products)
- * productSearchService (to search for products - e.g. used on category listing page)
- * categoryService (for Flamingo Commerce "category" module - to receive categorys and category trees). To use the Adapter you need to add the main `CategoryModule` to your bootstrap.
- * searchService (for Flamingo Commerce "search" module allowing searching for products documents). To use the Adapters you need to add the main `SearchModule` to your bootstrap.
+
+* productService (to retrieve single products)
+* productSearchService (to search for products - e.g. used on category listing page)
+* categoryService (for Flamingo Commerce "category" module - to receive categorys and category trees). To use the
```
Comment (el0c1n) [inline commercesearch/Readme.md]:
```suggestion
* categoryService (for Flamingo Commerce "category" module - to receive categories and category trees). To use the
```

File: commercesearch/Readme.md [hunk 5]
```diff
@@ -3,59 +3,69 @@
...
 The indexing (loading) of products is done by implementing and registering the `IndexUpdater` interface:
+
 ```go
 
-	//IndexUpdater - interface to Load products in a Index - secondary port
-    IndexUpdater interface {
-        //Indexer method that is called with an initialized Indexer. The passed Indexer provides helpers to update the Repository
-       Index(ctx context.Context, rep *Indexer) error
-    }
+// IndexUpdater interface to Load products in a Index - secondary port
```
Comment (el0c1n) [inline commercesearch/Readme.md]:
```suggestion
// IndexUpdater interface to load products in an Index - secondary port
```

File: commercesearch/domain/indexing.go [hunk 6]
```diff
@@ -27,15 +28,15 @@ type (
 		batchCatQueue      []product.CategoryTeaser
 	}
 
-	//CategoryTreeBuilder helper to build category tree
+	// CategoryTreeBuilder helper to build category tree
 	CategoryTreeBuilder struct {
-		//rootCategory - this is the link into the tree that is going to be build
+		// rootCategory - this is the link into the tree that is going to be build
```
Comment (el0c1n) [inline commercesearch/domain/indexing.go]:
```suggestion
		// rootCategory - this is the link into the tree that is going to be built
```

File: commercesearch/domain/indexing.go [hunk 7]
```diff
@@ -27,15 +28,15 @@ type (
...
 		rootCategory *categoryDomain.TreeData
 
-		//categorTreeIndex - the link into the treenode - is build
-		categorTreeIndex map[string]*categoryDomain.TreeData
+		// categoryTreeIndex - the link into the treenode - is build
```
Comment (el0c1n) [inline commercesearch/domain/indexing.go]:
```suggestion
		// categoryTreeIndex - the link into the treenode - is built
```

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go [hunk 8]
```diff
@@ -107,7 +109,7 @@ func (r *InMemoryProductRepository) UpdateProducts(_ context.Context, products [
 		}
 		r.marketplaceCodeIndex[product.BaseData().MarketPlaceCode] = product
```
Comment (el0c1n) [inline commercesearch/infrastructure/productsearch/inmemoryrepository.go]: maybe use the marketPlaceCode variable for indexing instead

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go [hunk 9]
```diff
@@ -20,19 +22,19 @@ type (
 
 	// InMemoryProductRepository serves as a Repository of Products held in memory
 	InMemoryProductRepository struct {
-		//marketplaceCodeIndex - index to get products from marketplaceCode
+		// marketplaceCodeIndex index to get products from marketplaceCode
 		marketplaceCodeIndex map[string]productDomain.BasicProduct
 
-		//attributeReverseIndex - index to get products from attribute
+		// attributeReverseIndex index to get products from attribute
 		attributeReverseIndex map[string]map[string][]string
```
Comment (el0c1n) [inline commercesearch/infrastructure/productsearch/inmemoryrepository.go:29]: this index is a mapping from attribute to marketplace code, right? I'm not sure if it's helpful to add this to the naming or alternatively to add this to the comments that it maps to the product via the marketplace code

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go [hunk 10]
```diff
@@ -20,19 +22,19 @@ type (
...
 
-		//productsByCategoriesReverseIndex - index to get products by categoryCode
+		// productsByCategoriesReverseIndex index to get products by categoryCode
 		productsByCategoriesReverseIndex map[string][]string
```
Comment (el0c1n) [inline commercesearch/infrastructure/productsearch/inmemoryrepository.go:32]: same as above

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go [hunk 11]
```diff
@@ -107,7 +109,7 @@ func (r *InMemoryProductRepository) UpdateProducts(_ context.Context, products [
 		}
 		r.marketplaceCodeIndex[product.BaseData().MarketPlaceCode] = product
 
-		//Now add product to category indexes:
+		// Now add product to category indexes:
 		if r.productsByCategoriesReverseIndex == nil {
```
Comment (el0c1n) [inline commercesearch/infrastructure/productsearch/inmemoryrepository.go]: could be moved to a function to make this one slimmer and more readable

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go [hunk 12]
```diff
@@ -121,7 +123,7 @@ func (r *InMemoryProductRepository) UpdateProducts(_ context.Context, products [
 			}
 		}
 
-		//Now fill the reverse index for all products attributes:
+		// Now fill the reverse index for all products attributes:
 		if r.attributeReverseIndex == nil {
```
Comment (el0c1n) [inline commercesearch/infrastructure/productsearch/inmemoryrepository.go]: could be moved to a function to make this one slimmer and more readable

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go [hunk 13]
```diff
@@ -310,33 +312,33 @@ func (s *marketPlaceCodeSet) intersection(set2 []string) {
...
 func (r *InMemoryProductRepository) addCategoryPath(currentTreeNode *categoryDomain.TreeData, treeNodeToAdd *categoryDomain.TreeData) *categoryDomain.TreeData {
-	if r.categorTreeIndex == nil {
-		r.categorTreeIndex = make(map[string]*categoryDomain.TreeData)
+	if r.categoryTreeIndex == nil {
+		r.categoryTreeIndex = make(map[string]*categoryDomain.TreeData)
 	}
-	//if its the first node then make as current node
+	// if its the first node then make as current node
 	if currentTreeNode == nil {
 		clone := *treeNodeToAdd
```
Comment (el0c1n) [inline commercesearch/infrastructure/productsearch/inmemoryrepository.go:337]: is this really a clone? seems confusing

File: csvindexing/infrastructure/commercesearch/indexing.go [hunk 14]
```diff
@@ -38,47 +40,67 @@ var (
...
 		if err != nil {
 			return errors.New(err.Error() + " / File: " + f.categoryCsvFile)
 		}
 		for _, row := range catrows {
 			f.categoryTreeBuilder.AddCategoryData(row["code"], row["label-"+f.locale], row["parent"])
 		}
 		tree, err = f.categoryTreeBuilder.BuildTree()
-		//fmt.Printf("\n %#v",tree)
-		//printTree(tree,"")
+		// fmt.Printf("\n %#v",tree)
```
Comment (el0c1n) [inline csvindexing/infrastructure/commercesearch/indexing.go]: are these still needed?

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go (hunk 9 above)
Comment (carstendietrich) [reply commercesearch/infrastructure/productsearch/inmemoryrepository.go:29]: updated comment

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go (hunk 8 above)
Comment (carstendietrich) [reply commercesearch/infrastructure/productsearch/inmemoryrepository.go]: good point

File: commercesearch/infrastructure/productsearch/inmemoryrepository.go (hunk 13 above)
Comment (carstendietrich) [reply commercesearch/infrastructure/productsearch/inmemoryrepository.go:337]: as soon as you dereference a struct and assign it to a new variable you basically copy the struct, this only works for simple structs that don't hold pointers themselves.

Comment (carstendietrich) [reply commercesearch/infrastructure/productsearch/inmemoryrepository.go:337]: https://play.golang.org/p/HOkZgosR4_-

**Review by el0c1n — APPROVED** [review]

**Pull request merged**